client/src/components/organizations/OrganizationSwitchPopover.vue

```diff
@@ -112,11 +128,13 @@ onMounted(async () => {
 
 async function onOrganizationClick(org: ConnectedOrganization): Promise<void> {
   if (org.handle !== getConnectionHandle()) {
-    await popoverController.dismiss({
-      handle: org.handle,
-    });
+    await popoverController.dismiss({ handle: org.handle }, MsModalResult.Confirm);
```

Comment:
Web is a special case: everything is mocked but the `listDevices`. So it displays "real" devices, but when you try to log with them, it's all fake. No real login, no real workspaces, files, users. It uses a fake handle (the same for all devices).

It doesn't work because you're trying to go from `/42/workspaces` to `/42/workspaces`, which is exactly the same URL, so no reason to navigate.

It could be possible to use different handles and all that but it's web, nothing is supposed to work. When it seems like it's doing something, it's only because we faked it to look that way.